### PR TITLE
identity: added isSignedWithPurpose support

### PR DESCRIPTION
**Pull request by manuelpolzhofer**
please review #729 first
- added support for isSignedWithPurpose
- added function to util package to convert an address to 32 bytes
  - starting with 12 zeros on the left

### CODE REVIEW

File: utils/tools.go [hunk 1]
```diff
@@ -65,6 +65,19 @@ func CheckMultiple32BytesFilled(b []byte, bs ...[]byte) bool {
 	return true
 }
 
+// AddressTo32Bytes converts an address to 32 a byte array
+// The length of an address is 20 bytes. First 12 bytes are filled with zeros.
+func AddressTo32Bytes(address common.Address) [32]byte {
```
Comment (mikiquantum) [inline utils/tools.go]: This sounds like an interesting function to have just passed the `bytes` and pad up front will all the zeros (not just address specific). 

Just out of curiosity why padding upfront?

Comment (mikiquantum) [reply utils/tools.go]: Can we use shifting bytes instead of for loop? Sounds a bit more efficient.

Comment (manuelpolzhofer) [reply utils/tools.go]: In Solidity if you convert an address to byte32 the padding happens in the left.

Comment (vedhavyas) [reply utils/tools.go]: You can instead do something like this

```
data := append(make([]byte, 12, 12), address.Bytes()...)
return SliceToByte32(data)
```

This is much efficient than for loop IMO

Comment (manuelpolzhofer) [reply utils/tools.go]: hehe. I tried it already at did a benchmark test.
The current solution is faster

**Pull request merged**